Src/PerformanceCollector/Shared/QuickPulseTelemetryProcessor.cs

```diff
@@ -157,13 +165,78 @@ public void Process(ITelemetry telemetry)
             }
             finally
             {
-                if (letItemThrough)
+                if (letTelemetryThrough)
                 {
                     this.Next.Process(telemetry);
                 }
             }
         }
 
+        private static ITelemetryDocument ConvertTelemetryToTelemetryDocument(
+            RequestTelemetry telemetryAsRequest,
+            DependencyTelemetry telemetryAsDependency,
+            ExceptionTelemetry telemetryAsException)
```

Comment:
Taking a single ITelemetry would make a cleaner interface and not require changes in future if more types are supported.